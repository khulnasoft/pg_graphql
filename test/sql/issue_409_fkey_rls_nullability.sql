begin;

    create table account(
        id int primary key
    );

    create table address(
        id int primary key,
        account_id int not null references account(id)
    );

    select jsonb_pretty(
        graphql.resolve($$
        {
          __type(name: "Account") {
            kind
            fields {
              name
              type {
                name
                kind
                ofType { name }
              }
            }
          }
        }
        $$)
    );

    select jsonb_pretty(
        graphql.resolve($$
        {
          __type(name: "Address") {
            kind
            fields {
              name
              type {
                name
                kind
                ofType { name }
              }
            }
          }
        }
        $$)
    );

    alter table account enable row level security;

    select jsonb_pretty(
        graphql.resolve($$
        {
          __type(name: "Account") {
            kind
            fields {
              name
              type {
                name
                kind
                ofType { name }
              }
            }
          }
        }
        $$)
    );

    select jsonb_pretty(
        graphql.resolve($$
        {
          __type(name: "Address") {
            kind
            fields {
              name
              type {
                name
                kind
                ofType { name }
              }
            }
          }
        }
        $$)
    );

    alter table account disable row level security;
    alter table address enable row level security;

    select jsonb_pretty(
        graphql.resolve($$
        {
          __type(name: "Account") {
            kind
            fields {
              name
              type {
                name
                kind
                ofType { name }
              }
            }
          }
        }
        $$)
    );

    select jsonb_pretty(
        graphql.resolve($$
        {
          __type(name: "Address") {
            kind
            fields {
              name
              type {
                name
                kind
                ofType { name }
              }
            }
          }
        }
        $$)
    );

rollback;
