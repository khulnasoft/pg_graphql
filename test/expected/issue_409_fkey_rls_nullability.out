begin;
    create table account(
        id int primary key
    );
    create table address(
        id int primary key,
        account_id int not null references account(id)
    );
    select jsonb_pretty(
        graphql.resolve($$
        {
          __type(name: "Account") {
            kind
            fields {
              name
              type {
                name
                kind
                ofType { name }
              }
            }
          }
        }
        $$)
    );
                     jsonb_pretty                     
------------------------------------------------------
 {                                                   +
     "data": {                                       +
         "__type": {                                 +
             "kind": "OBJECT",                       +
             "fields": [                             +
                 {                                   +
                     "name": "nodeId",               +
                     "type": {                       +
                         "kind": "NON_NULL",         +
                         "name": null,               +
                         "ofType": {                 +
                             "name": "ID"            +
                         }                           +
                     }                               +
                 },                                  +
                 {                                   +
                     "name": "id",                   +
                     "type": {                       +
                         "kind": "NON_NULL",         +
                         "name": null,               +
                         "ofType": {                 +
                             "name": "Int"           +
                         }                           +
                     }                               +
                 },                                  +
                 {                                   +
                     "name": "addressCollection",    +
                     "type": {                       +
                         "kind": "OBJECT",           +
                         "name": "AddressConnection",+
                         "ofType": null              +
                     }                               +
                 }                                   +
             ]                                       +
         }                                           +
     }                                               +
 }
(1 row)

    select jsonb_pretty(
        graphql.resolve($$
        {
          __type(name: "Address") {
            kind
            fields {
              name
              type {
                name
                kind
                ofType { name }
              }
            }
          }
        }
        $$)
    );
                 jsonb_pretty                  
-----------------------------------------------
 {                                            +
     "data": {                                +
         "__type": {                          +
             "kind": "OBJECT",                +
             "fields": [                      +
                 {                            +
                     "name": "nodeId",        +
                     "type": {                +
                         "kind": "NON_NULL",  +
                         "name": null,        +
                         "ofType": {          +
                             "name": "ID"     +
                         }                    +
                     }                        +
                 },                           +
                 {                            +
                     "name": "id",            +
                     "type": {                +
                         "kind": "NON_NULL",  +
                         "name": null,        +
                         "ofType": {          +
                             "name": "Int"    +
                         }                    +
                     }                        +
                 },                           +
                 {                            +
                     "name": "accountId",     +
                     "type": {                +
                         "kind": "NON_NULL",  +
                         "name": null,        +
                         "ofType": {          +
                             "name": "Int"    +
                         }                    +
                     }                        +
                 },                           +
                 {                            +
                     "name": "account",       +
                     "type": {                +
                         "kind": "NON_NULL",  +
                         "name": null,        +
                         "ofType": {          +
                             "name": "Account"+
                         }                    +
                     }                        +
                 }                            +
             ]                                +
         }                                    +
     }                                        +
 }
(1 row)

    alter table account enable row level security;
    select jsonb_pretty(
        graphql.resolve($$
        {
          __type(name: "Account") {
            kind
            fields {
              name
              type {
                name
                kind
                ofType { name }
              }
            }
          }
        }
        $$)
    );
                     jsonb_pretty                     
------------------------------------------------------
 {                                                   +
     "data": {                                       +
         "__type": {                                 +
             "kind": "OBJECT",                       +
             "fields": [                             +
                 {                                   +
                     "name": "nodeId",               +
                     "type": {                       +
                         "kind": "NON_NULL",         +
                         "name": null,               +
                         "ofType": {                 +
                             "name": "ID"            +
                         }                           +
                     }                               +
                 },                                  +
                 {                                   +
                     "name": "id",                   +
                     "type": {                       +
                         "kind": "NON_NULL",         +
                         "name": null,               +
                         "ofType": {                 +
                             "name": "Int"           +
                         }                           +
                     }                               +
                 },                                  +
                 {                                   +
                     "name": "addressCollection",    +
                     "type": {                       +
                         "kind": "OBJECT",           +
                         "name": "AddressConnection",+
                         "ofType": null              +
                     }                               +
                 }                                   +
             ]                                       +
         }                                           +
     }                                               +
 }
(1 row)

    select jsonb_pretty(
        graphql.resolve($$
        {
          __type(name: "Address") {
            kind
            fields {
              name
              type {
                name
                kind
                ofType { name }
              }
            }
          }
        }
        $$)
    );
                jsonb_pretty                 
---------------------------------------------
 {                                          +
     "data": {                              +
         "__type": {                        +
             "kind": "OBJECT",              +
             "fields": [                    +
                 {                          +
                     "name": "nodeId",      +
                     "type": {              +
                         "kind": "NON_NULL",+
                         "name": null,      +
                         "ofType": {        +
                             "name": "ID"   +
                         }                  +
                     }                      +
                 },                         +
                 {                          +
                     "name": "id",          +
                     "type": {              +
                         "kind": "NON_NULL",+
                         "name": null,      +
                         "ofType": {        +
                             "name": "Int"  +
                         }                  +
                     }                      +
                 },                         +
                 {                          +
                     "name": "accountId",   +
                     "type": {              +
                         "kind": "NON_NULL",+
                         "name": null,      +
                         "ofType": {        +
                             "name": "Int"  +
                         }                  +
                     }                      +
                 },                         +
                 {                          +
                     "name": "account",     +
                     "type": {              +
                         "kind": "OBJECT",  +
                         "name": "Account", +
                         "ofType": null     +
                     }                      +
                 }                          +
             ]                              +
         }                                  +
     }                                      +
 }
(1 row)

    alter table account disable row level security;
    alter table address enable row level security;
    select jsonb_pretty(
        graphql.resolve($$
        {
          __type(name: "Account") {
            kind
            fields {
              name
              type {
                name
                kind
                ofType { name }
              }
            }
          }
        }
        $$)
    );
                     jsonb_pretty                     
------------------------------------------------------
 {                                                   +
     "data": {                                       +
         "__type": {                                 +
             "kind": "OBJECT",                       +
             "fields": [                             +
                 {                                   +
                     "name": "nodeId",               +
                     "type": {                       +
                         "kind": "NON_NULL",         +
                         "name": null,               +
                         "ofType": {                 +
                             "name": "ID"            +
                         }                           +
                     }                               +
                 },                                  +
                 {                                   +
                     "name": "id",                   +
                     "type": {                       +
                         "kind": "NON_NULL",         +
                         "name": null,               +
                         "ofType": {                 +
                             "name": "Int"           +
                         }                           +
                     }                               +
                 },                                  +
                 {                                   +
                     "name": "addressCollection",    +
                     "type": {                       +
                         "kind": "OBJECT",           +
                         "name": "AddressConnection",+
                         "ofType": null              +
                     }                               +
                 }                                   +
             ]                                       +
         }                                           +
     }                                               +
 }
(1 row)

    select jsonb_pretty(
        graphql.resolve($$
        {
          __type(name: "Address") {
            kind
            fields {
              name
              type {
                name
                kind
                ofType { name }
              }
            }
          }
        }
        $$)
    );
                 jsonb_pretty                  
-----------------------------------------------
 {                                            +
     "data": {                                +
         "__type": {                          +
             "kind": "OBJECT",                +
             "fields": [                      +
                 {                            +
                     "name": "nodeId",        +
                     "type": {                +
                         "kind": "NON_NULL",  +
                         "name": null,        +
                         "ofType": {          +
                             "name": "ID"     +
                         }                    +
                     }                        +
                 },                           +
                 {                            +
                     "name": "id",            +
                     "type": {                +
                         "kind": "NON_NULL",  +
                         "name": null,        +
                         "ofType": {          +
                             "name": "Int"    +
                         }                    +
                     }                        +
                 },                           +
                 {                            +
                     "name": "accountId",     +
                     "type": {                +
                         "kind": "NON_NULL",  +
                         "name": null,        +
                         "ofType": {          +
                             "name": "Int"    +
                         }                    +
                     }                        +
                 },                           +
                 {                            +
                     "name": "account",       +
                     "type": {                +
                         "kind": "NON_NULL",  +
                         "name": null,        +
                         "ofType": {          +
                             "name": "Account"+
                         }                    +
                     }                        +
                 }                            +
             ]                                +
         }                                    +
     }                                        +
 }
(1 row)

rollback;
